### PR TITLE
NO-ISSUE Extend Hive integration doc

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by rollandf** [comment]
Maybe I will add some Day2 explanation

File: docs/Kube-API-examples.md [hunk 1]
```diff
@@ -1,4 +1,48 @@
...
+In the Agent, the user can specify the hostname, role, installation disk and more.
+Also, the host hardware inventory and statuses are available.
+
+Note that if the Agent is not Approved, it will not be part of the installation.
+
+Here how to approve an Agent:
+
+`kubectl -n assisted-installer patch agents.adi.io.my.domain 120af504-d88e-46bd-bec2-b8b261db3b01 -p '{"spec":{"approved":true}}' --type merge`
+
+## Day 2 worker
```
Comment (nmagnezi) [inline docs/Kube-API-examples.md:39]: Do we have a day2 doc to reference here?

File: docs/Kube-API-examples.md [hunk 2]
```diff
@@ -1,4 +1,48 @@
...
+Once the cluster is installed, the ClusterDeployement is set to Installed and secrets for kubeconfig and credentials are created.
+In the Assisted Service, the original cluster is deleted and a Day 2 cluster is created instead.
+
+Additional nodes can be added by booting from the new generated ISO. Each additional host will start installation once the Agent is Approved and the Host is in Known state.
```
Comment (nmagnezi) [inline docs/Kube-API-examples.md:43]:
```suggestion
Additional nodes can be added by booting from the new generated ISO. Each additional host will start installation once the Agent is Approved and the Host is in known state.
```

File: docs/Kube-API-examples.md [hunk 3]
```diff
@@ -1,4 +1,48 @@
...
+
+### [InstallEnv](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1alpha1/installenv_types.go)
+The InstallEnv CRD represents the configuration needed to create the discovery ISO.
+The user can specify proxy settings, ignition overrides and specify NMState labels.
+
+When the ISO is ready, an URL will be available in the CR.
+
+
+### [NMStateConfig](https://github.com/openshift/assisted-service/blob/master/internal/controller/api/v1alpha1/nmstate_config_types.go)
+The NMStateConfig contains network configuration that will applied on the hosts.
```
Comment (nmagnezi) [inline docs/Kube-API-examples.md:24]: NMState reference https://github.com/nmstate/kubernetes-nmstate/blob/master/README.md

**Review by nmagnezi — COMMENTED** [review]
very nice!

File: docs/Kube-API-examples.md (hunk 3 above)
Comment (rollandf) [reply docs/Kube-API-examples.md]: This repo is for kubernetes. I added referebce to https://github.com/nmstate/nmstate

File: docs/Kube-API-examples.md (hunk 1 above)
Comment (rollandf) [reply docs/Kube-API-examples.md:39]: I am not aware of any.

File: docs/Kube-API-examples.md (hunk 2 above)
Comment (rollandf) [reply docs/Kube-API-examples.md]: done

**Review by filanov — APPROVED** [review]

**Pull request merged**